src/modules/stilling/rapporter-annonse/components/ReportAd.jsx

```diff
@@ -1,77 +1,57 @@
+    Link as AkselLink,
+    LinkPanel,
+    Textarea,
+    Fieldset,
+    CheckboxGroup,
+    VStack,
+    BodyShort,
+} from "@navikt/ds-react";
 import logAmplitudeEvent from "../../../common/tracking/amplitude";
 import UserAPI from "../../../common/api/UserAPI";
-import H1WithAutoFocus from "../../../common/components/h1WithAutoFocus/H1WithAutoFocus";
 
-const violationCategories = [
+const reportCategories = [
     { label: "Diskriminerende innhold", key: "discrimination" },
     { label: "Det er markedsføring", key: "marketing" },
-    { label: "Annet", key: "other" },
-];
-
-const scamCategories = [
     { label: "Falsk stillingannonse og arbeidsgiver", key: "fake" },
     { label: "Krever betaling for å søke stilling", key: "payment" },
     { label: "Ber om kredittinfo og/eller BankID", key: "creditInfo" },
     { label: "Annet", key: "other" },
 ];
 
-function ReportAd({ id }) {
+function ReportAd({ ad }) {
     const [error, setError] = useState(false);
+    const [isSubmittingForm, setIsSubmittingForm] = useState(false);
+    const [validationError, setValidationError] = useState(null);
     const [finished, setFinished] = useState(false);
-    const [violation, setViolation] = useState(false);
-    const [violationCategory, setViolationCategory] = useState(null);
-    const [scam, setScam] = useState(false);
-    const [scamCategory, setScamCategory] = useState(null);
+    const [category, setCategory] = useState(null);
     const [description, setDescription] = useState("");
-    const [descriptionLabel, setDescriptionLabel] = useState("Beskrivelse - må fylles ut");
-
-    function handleDescriptionChange(e) {
-        setDescription(e.target.value);
-    }
+    const categoryFieldError = "Du må velge minst én grunn til at annonsen bryter rettningslinjene";
+    const messageFieldError = "Du har brukt for mange tegn";
 
-    function handleViolationCheck() {
-        setViolation(!violation);
-        setScam(false);
-        setViolationCategory(null);
-        setScamCategory(null);
-        setDescriptionLabel("Beskriv regelbruddet - må fylles ut");
-    }
-
-    function handleViolationCategoryCheck(e) {
-        setViolationCategory(e.target.value);
-    }
-
-    function handleScamCheck() {
-        setScam(!scam);
-        setViolation(false);
-        setViolationCategory(null);
-        setScamCategory(null);
-        setDescriptionLabel("Beskriv svindelen - må fylles ut");
-    }
-
-    function handleScamCategoryCheck(e) {
-        setScamCategory(e.target.value);
-    }
-
-    async function handleSendTip() {
-        const category = violation ? "Regelbrudd" : "Mistanke om svindel";
-        const subCategory = violation
-            ? violationCategories.filter((c) => c.key === violationCategory)[0].label
-            : scamCategories.filter((c) => c.key === scamCategory)[0].label;
-        const title = `En stilling har blitt rapportert for ${category.toLowerCase()}`;
+    const submitForm = async () => {
```

Comment:
Her kan man bruke formData, som er web standard, i stedet for å ha data i useState("something").
Hvis du gir alle skjemaelementer name, f.eks name="category", så vil du få disse dataene i formData.
```
  function onFormSybmit(e) {
        e.preventDefault(); // Prevent form submit by browser

        const formData = new FormData(e.target);
        const category = formData.get("category");
        
        postAdReporting(category);
    }
```

Reply:
Så kan du trekke postAdReporting ut som en egen funksjon og flytte den opp i page.js (som da er veldig client-side, next.js vil ha en egen page.js som gjør det anderledes. Men ved å flytte alt med state og fetch opp i page.js så blir komponentene lengre ned kjørbare begge steder)